### DISCUSSION TITLE
Infinite scroll stop query if has no more pages

### DISCUSSION THREAD

**Discussion by t-prod** [livewire/livewire#5445] [1. Help]
Hi,

I'm developing an infinite scroll by emitting a loadMore emit when user scroll at the bottom of the page.
How can I stop doing this when pagination has no more pages ?

```
 window.onscroll = function(ev) {
        if ((window.innerHeight + window.pageYOffset) >= document.body.offsetHeight) {
            window.livewire.emit('loadMore');
        }
    };
```

``` 
public function loadMore()
{
    $this->limitPerPage = $this->limitPerPage + 10;
}
```

``` 
public function getQuery()
{
        $this->products = $this->products->paginate($this->limitPerPage);
}
```

I know there's an helper for this => $products->hasMorePages() when we do a paginate ?
How can I stop querying when there's no more results when I scroll down ?

Any ideas ?

Regards.

**Comment by t-prod** [comment]
Any ideas ? I would like to detect where it's not necessary to emit again cause we have no products to load anymore ;)

**Comment by joshhanley** [comment]
@t-prod maybe you can calculate how many pages are remaining in your `loadMore` method and then store that in a properly or dispatch a browser event with that value, so your script can access that, and check if there is more pages, then trigger load more, but if there isn't, then do nothing?

Hope this helps!
